internal/http/healthcheck_server.go

```diff
@@ -0,0 +1,61 @@
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"task-runner-launcher/internal/logs"
+	"time"
+)
+
+const (
+	defaultPort     = 5681
+	portEnvVar      = "N8N_LAUNCHER_HEALTCHECK_PORT"
+	healthCheckPath = "/healthz"
+	readTimeout     = 1 * time.Second
+	writeTimeout    = 1 * time.Second
+)
+
+func NewHealthCheckServer() *http.Server {
+	mux := http.NewServeMux()
+	mux.HandleFunc(healthCheckPath, handleHealthCheck)
+
+	return &http.Server{
+		Addr:         fmt.Sprintf(":%d", GetPort()),
+		Handler:      mux,
+		ReadTimeout:  readTimeout,
+		WriteTimeout: writeTimeout,
+	}
+}
+
+func handleHealthCheck(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	res := struct {
+		Status string `json:"status"`
+	}{Status: "ok"}
+
+	if err := json.NewEncoder(w).Encode(res); err != nil {
+		logs.Logger.Printf("Failed to encode health check response: %v", err)
```

Comment:
Shouldn't we log with error level?

Reply:
Yeah back then this was waiting for the logging PR to be merged, same below in `GetPort` 👍🏻